src/Config.php

```diff
@@ -437,9 +440,7 @@ public function onProduction() : bool
             return false;
         }
 
-        $prodBranch = $this->onDedicated() ? 'production' : 'master';
-
-        return $this->getValue('BRANCH') == $prodBranch;
+        return $this->getValue('ENVIRONMENT_TYPE') == 'production';
```

Comment:
It looks like some legacy environments will not have that variable, so we need to keep the current logic as backwards compatibility here.